[Bug]: panic: runtime error: invalid memory address or nil pointer dereference (GetVolumeRebateStats)

### Problem encountered

Local artifacts
[vega-sim-olwlknpf.zip](https://github.com/user-attachments/files/16615475/vega-sim-olwlknpf.zip)

```
2024-08-14T16:21:44.664+0100	INFO	core	node/node.go:99	starting vega	{"version": "v0.78.0-dev", "commit-hash": "f8fee6448d5f5d439869a8fef6b7b225da4bf055"}
```

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x10237d2b4]

goroutine 235134 [running]:
code.vegaprotocol.io/vega/datanode/api.(*TradingDataServiceV2).GetVolumeRebateStats(0x14002a2a000, {0x104200ac8, 0x14000c6b140}, 0x14003b5bd80)
	/Users/caed/Projects/vega-market-sim/extern/vega/datanode/api/trading_data_v2.go:5490 +0xd4
code.vegaprotocol.io/vega/protos/data-node/api/v2._TradingDataService_GetVolumeRebateStats_Handler.func1({0x104200ac8, 0x14000c6b140}, {0x103ed7980?, 0x14003b5bd80})
	/Users/caed/Projects/vega-market-sim/extern/vega/protos/data-node/api/v2/trading_data_grpc.pb.go:5320 +0x74
code.vegaprotocol.io/vega/datanode/ratelimit.(*RateLimit).GRPCInterceptor(0x14001ed8f20?, {0x104200ac8?, 0x14000c6b140?}, {0x103ed7980?, 0x14003b5bd80?}, 0x5000104?, 0x0?)
	/Users/caed/Projects/vega-market-sim/extern/vega/datanode/ratelimit/ratelimit.go:150 +0xcc
google.golang.org/grpc.getChainUnaryHandler.func1({0x104200ac8, 0x14000c6b140}, {0x103ed7980, 0x14003b5bd80})
	/Users/caed/go/pkg/mod/google.golang.org/grpc@v1.60.0/server.go:1192 +0xa0
code.vegaprotocol.io/vega/datanode/api.(*GRPCServer).Start.headersInterceptor.func4({0x104200ac8, 0x14000c6b140}, {0x103ed7980, 0x14003b5bd80}, 0x14000a0fae0?, 0x14003b5be00)
	/Users/caed/Projects/vega-market-sim/extern/vega/datanode/api/server.go:469 +0x328
google.golang.org/grpc.getChainUnaryHandler.func1({0x104200ac8, 0x14000c6b140}, {0x103ed7980, 0x14003b5bd80})
	/Users/caed/go/pkg/mod/google.golang.org/grpc@v1.60.0/server.go:1192 +0xa0
code.vegaprotocol.io/vega/datanode/api.(*GRPCServer).Start.(*GRPCServer).remoteAddrInterceptor.func3({0x104200ac8, 0x14000c6b0e0}, {0x103ed7980, 0x14003b5bd80}, 0x14000a0fae0, 0x14003b5bdc0)
	/Users/caed/Projects/vega-market-sim/extern/vega/datanode/api/server.go:425 +0x9c
google.golang.org/grpc.NewServer.chainUnaryServerInterceptors.chainUnaryInterceptors.func1({0x104200ac8, 0x14000c6b0e0}, {0x103ed7980, 0x14003b5bd80}, 0x140029499a8?, 0x100db42f8?)
	/Users/caed/go/pkg/mod/google.golang.org/grpc@v1.60.0/server.go:1183 +0x88
code.vegaprotocol.io/vega/protos/data-node/api/v2._TradingDataService_GetVolumeRebateStats_Handler({0x104187240?, 0x14002a2a000}, {0x104200ac8, 0x14000c6b0e0}, 0x14002ea6e80, 0x14002a12120)
	/Users/caed/Projects/vega-market-sim/extern/vega/protos/data-node/api/v2/trading_data_grpc.pb.go:5322 +0x12c
google.golang.org/grpc.(*Server).processUnaryRPC(0x14002a24000, {0x104200ac8, 0x14003dcba40}, {0x104216480, 0x14000bfb6c0}, 0x140025cd560, 0x14000ba2990, 0x1061a5638, 0x0)
	/Users/caed/go/pkg/mod/google.golang.org/grpc@v1.60.0/server.go:1372 +0xb8c
google.golang.org/grpc.(*Server).handleStream(0x14002a24000, {0x104216480, 0x14000bfb6c0}, 0x140025cd560)
	/Users/caed/go/pkg/mod/google.golang.org/grpc@v1.60.0/server.go:1783 +0xc4c
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	/Users/caed/go/pkg/mod/google.golang.org/grpc@v1.60.0/server.go:1016 +0x5c
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 442
	/Users/caed/go/pkg/mod/google.golang.org/grpc@v1.60.0/server.go:1027 +0x138
```

### Observed behaviour

-

### Expected behaviour

-

### Steps to reproduce

```bash
replay
```


### Software version

f8fee6448d5f5d439869a8fef6b7b225da4bf055

### Failing test

vega-market-sim fuzz test

### Jenkins run

_No response_

### Configuration used

```shell
{
  "genesis_time": "2023-05-09T10:00:00.000000000Z",
  "chain_id": "CUSTOM",
  "initial_height": "0",
  "consensus_params": {
    "block": {
      "max_bytes": "22020096",
      "max_gas": "-1",
      "time_iota_ms": "1"
    },
    "evidence": {
      "max_age_num_blocks": "100000",
      "max_age_duration": "172800000000000",
      "max_bytes": "1048576"
    },
    "validator": {
      "pub_key_types": ["ed25519"]
    },
    "version": {}
  },
  "validators": [
    {
      "address": "7DFD3744A773C166B36F170E4F5EF3E998703F4E",
      "pub_key": {
        "type": "tendermint/PubKeyEd25519",
        "value": "2lPO6I0p5uCBZzeoA2+ECJ2+57LtLVMjXpxH5za9TvE="
      },
      "power": "10",
      "name": ""
    }
  ],
  "app_hash": "",
  "app_state": {
    "assets": {
      "0000000000000000000000000000000000000000000000000000000000000000": {
        "name": "VOTE",
        "symbol": "VOTE",
        "decimals": 5,
        "quantum": "1",
        "min_lp_stake": "1",
        "source": {
          "builtin_asset": {
            "max_faucet_amount_mint": "100000000000000000"
          }
        }
      }
    },
    "validators": {
      "6NemqXSfN4zfZvAQeIANflEFqQ98izX9C0KLI2dITSY=": {
        "id": "c2b0ab69fb1bd19862c04df401920ea62d3e0bce41a50148240f42017824d17a",
        "vega_pub_key": "dfcf0fa8d67cfa11b3a511e320f3c47807a5e0735cc06dd9c2bc00f7ccc67117",
        "vega_pub_key_index": 1,
        "ethereum_address": "0x9e5BEEC6E56B28cCbd02864840B0f1e0125e42Ce",
        "tm_pub_key": "6NemqXSfN4zfZvAQeIANflEFqQ98izX9C0KLI2dITSY=",
        "info_url": "",
        "country": "",
        "name": "",
        "avatar_url": ""
      }
    },
    "network": {
      "replay_attack_threshold": 150
    },
    "network_parameters": {
      "blockchains.ethereumConfig": "{\"network_id\": \"3\", \"chain_id\": \"3\", \"collateral_bridge_contract\": { \"address\": \"0xa6F1E140daC13002Dfd9789D6dBA59117c717D7a\" }, \"confirmations\": 50, \"staking_bridge_contract\": { \"address\": \"0xfce2CC92203A266a9C8e67461ae5067c78f67235\", \"deployment_block_height\": 11001702}, \"multisig_control_contract\": {\"address\": \"0xCF6d41235911184fe6F35D47207813bFF3B91601\", \"deployment_block_height\": 12710009 } }",
      "blockchains.evmBridgeConfigs": "{\"configs\":[{\"network_id\": \"3\", \"chain_id\": \"3\", \"collateral_bridge_contract\": { \"address\": \"0xa6F1E140daC13002Dfd9789D6dBA59117c717D7a\" }, \"confirmations\": 50, \"multisig_control_contract\": {\"address\": \"0xCF6d41235911184fe6F35D47207813bFF3B91601\", \"deployment_block_height\": 12710009 } }]}",
      "blockchains.ethereumRpcAndEvmCompatDataSourcesConfig": "{\"configs\": []}",
      "governance.proposal.asset.maxClose": "8760h0m0s",
      "governance.proposal.asset.maxEnact": "8760h0m0s",
      "governance.proposal.asset.minClose": "1s",
      "governance.proposal.asset.minEnact": "1s",
      "governance.proposal.asset.minProposerBalance": "1",
      "governance.proposal.asset.minVoterBalance": "1",
      "governance.proposal.asset.requiredMajority": "0.66",
      "governance.proposal.asset.requiredParticipation": "0.00001",
      "governance.proposal.freeform.maxClose": "8760h0m0s",
      "governance.proposal.freeform.minClose": "1s",
      "governance.proposal.freeform.minProposerBalance": "1",
      "governance.proposal.freeform.minVoterBalance": "1",
      "governance.proposal.freeform.requiredMajority": "0.66",
      "governance.proposal.freeform.requiredParticipation": "0.00001",
      "governance.proposal.market.maxClose": "8760h0m0s",
      "governance.proposal.market.maxEnact": "8760h0m0s",
      "governance.proposal.market.minClose": "1s",
      "governance.proposal.market.minEnact": "2s",
      "governance.proposal.market.minProposerBalance": "1",
      "governance.proposal.market.minVoterBalance": "1",
      "governance.proposal.market.requiredMajority": "0.66",
      "governance.proposal.market.requiredParticipation": "0.00001",
      "governance.proposal.updateAsset.maxClose": "720h",
      "governance.proposal.updateAsset.maxEnact": "720h",
      "governance.proposal.updateAsset.minClose": "1s",
      "governance.proposal.updateAsset.minEnact": "1s",
      "governance.proposal.updateAsset.minProposerBalance": "1",
      "governance.proposal.updateAsset.minVoterBalance": "1",
      "governance.proposal.updateAsset.requiredMajority": "0.66",
      "governance.proposal.updateAsset.requiredParticipation": "0.09",
      "governance.proposal.updateMarket.maxClose": "8760h0m0s",
      "governance.proposal.updateMarket.maxEnact": "8760h0m0s",
      "governance.proposal.updateMarket.minClose": "1s",
      "governance.proposal.updateMarket.minEnact": "1s",
      "governance.proposal.updateMarket.minProposerBalance": "1",
      "governance.proposal.updateMarket.minProposerEquityLikeShare": "0",
      "governance.proposal.updateMarket.minVoterBalance": "1",
      "governance.proposal.updateMarket.requiredMajority": "0.66",
      "governance.proposal.updateMarket.requiredMajorityLP": "0.66",
      "governance.proposal.updateMarket.requiredParticipation": "0.00001",
      "governance.proposal.updateMarket.requiredParticipationLP": "0",
      "governance.proposal.updateNetParam.maxClose": "8760h0m0s",
      "governance.proposal.updateNetParam.maxEnact": "8760h0m0s",
      "governance.proposal.updateNetParam.minClose": "1s",
      "governance.proposal.updateNetParam.minEnact": "1s",
      "governance.proposal.updateNetParam.minProposerBalance": "1",
      "governance.proposal.updateNetParam.minVoterBalance": "1",
      "governance.proposal.updateNetParam.requiredMajority": "0.5",
      "governance.proposal.updateNetParam.requiredParticipation": "0.00001",
      "governance.proposal.referralProgram.maxClose": "8760h0m0s",
      "governance.proposal.referralProgram.maxEnact": "8760h0m0s",
      "governance.proposal.referralProgram.minClose": "1s",
      "governance.proposal.referralProgram.minEnact": "1s",
      "governance.proposal.referralProgram.minProposerBalance": "1",
      "governance.proposal.referralProgram.minVoterBalance": "1",
      "governance.proposal.referralProgram.requiredMajority": "0.5",
      "governance.proposal.referralProgram.requiredParticipation": "0.00001",
      "governance.proposal.VolumeDiscountProgram.maxClose": "8760h0m0s",
      "governance.proposal.VolumeDiscountProgram.maxEnact": "8760h0m0s",
      "governance.proposal.VolumeDiscountProgram.minClose": "1s",
      "governance.proposal.VolumeDiscountProgram.minEnact": "1s",
      "governance.proposal.VolumeDiscountProgram.minProposerBalance": "1",
      "governance.proposal.VolumeDiscountProgram.minVoterBalance": "1",
      "governance.proposal.VolumeDiscountProgram.requiredMajority": "0.5",
      "governance.proposal.VolumeDiscountProgram.requiredParticipation": "0.00001",
      "governance.proposal.VolumeRebateProgram.maxClose": "8760h0m0s",
      "governance.proposal.VolumeRebateProgram.maxEnact": "8760h0m0s",
      "governance.proposal.VolumeRebateProgram.minClose": "1s",
      "governance.proposal.VolumeRebateProgram.minEnact": "1s",
      "governance.proposal.VolumeRebateProgram.minProposerBalance": "1",
      "governance.proposal.VolumeRebateProgram.minVoterBalance": "1",
      "governance.proposal.VolumeRebateProgram.requiredMajority": "0.5",
      "governance.proposal.VolumeRebateProgram.requiredParticipation": "0.00001",
      "governance.proposal.transfer.maxClose": "8760h0m0s",
      "governance.proposal.transfer.maxEnact": "8760h0m0s",
      "governance.proposal.transfer.minClose": "1s",
      "governance.proposal.transfer.minEnact": "1s",
      "governance.proposal.transfer.minProposerBalance": "1",
      "governance.proposal.transfer.minVoterBalance": "1",
      "governance.proposal.transfer.requiredMajority": "0.5",
      "governance.proposal.transfer.requiredParticipation": "0.00001",
      "limits.markets.maxPeggedOrders": "1500",
      "limits.markets.proposePerpetualEnabled": "1",
      "limits.markets.ammPoolEnabled": "1",
      "market.auction.maximumDuration": "168h",
      "market.auction.minimumDuration": "1s",
      "market.fee.factors.infrastructureFee": "0.0005",
      "market.fee.factors.makerFee": "0.0002",
      "market.fee.factors.treasuryFee": "0.0001",
      "market.fee.factors.buybackFee": "0.0001",
      "market.liquidity.bondPenaltyParameter": "0.1",
      "market.liquidity.earlyExitPenalty": "0.1",
      "market.liquidity.maximumLiquidityFeeFactorLevel": "0.03",
      "market.liquidity.minimum.probabilityOfTrading.lpOrders": "1e-6",
      "market.liquidity.probabilityOfTrading.tau.scaling": "10.0",
      "market.liquidity.providersFeeCalculationTimeStep": "1s",
      "market.liquidity.sla.nonPerformanceBondPenaltyMax": "0.5",
      "market.liquidity.sla.nonPerformanceBondPenaltySlope": "2",
      "market.liquidity.stakeToCcyVolume": "1.0",
      "market.liquidity.targetstake.triggering.ratio": "0.25",
      "market.liquidityProvision.minLpStakeQuantumMultiple": "5000",
      "market.liquidityProvision.shapes.maxSize": "100",
      "market.margin.scalingFactors": "{\"search_level\": 1.1, \"initial_margin\": 1.5, \"collateral_release\": 1.7}",
      "market.monitor.price.defaultParameters": "{\"triggers\": [{\"auction_extension\": 300, \"horizon\": 43200, \"probability\": \"0.9999999\"}] }",
      "market.stake.target.scalingFactor": "0.0001",
      "market.stake.target.timeWindow": "1h",
      "market.value.windowLength": "2h0m0s",
      "network.checkpoint.timeElapsedBetweenCheckpoints": "5m",
      "network.floatingPointUpdates.delay": "5m",
      "network.markPriceUpdateMaximumFrequency": "5s",
      "network.transaction.defaultgas": "1",
      "network.transactions.maxgasperblock": "3000000",
      "network.transactions.minBlockCapacity": "32",
      "network.validators.ersatz.multipleOfTendermintValidators": "0",
      "network.validators.ersatz.rewardFactor": "0.85",
      "network.validators.incumbentBonus": "0.05",
      "network.validators.minimumEthereumEventsForNewValidator": "3",
      "network.validators.multisig.numberOfSigners": "13",
      "network.validators.tendermint.number": "13",
      "reward.asset": "0000000000000000000000000000000000000000000000000000000000000000",
      "reward.staking.delegation.competitionLevel": "1.3",
      "reward.staking.delegation.delegatorShare": "0.8",
      "reward.staking.delegation.maxPayoutPerEpoch": "9890000000000000000000",
      "reward.staking.delegation.maxPayoutPerParticipant": "700000000000000000000",
      "reward.staking.delegation.minimumValidatorStake": "3000000000000000000000",
      "reward.staking.delegation.minValidators": "5",
      "reward.staking.delegation.optimalStakeMultiplier": "3.0",
      "reward.staking.delegation.payoutDelay": "0h",
      "reward.staking.delegation.payoutFraction": "1.0",
      "rewards.marketCreationQuantumMultiple": "1",
      "snapshot.interval.length": "300",
      "spam.pow.difficulty": "15",
      "spam.pow.hashFunction": "sha3_24_rounds",
      "spam.pow.increaseDifficulty": "0",
      "spam.pow.numberOfPastBlocks": "100",
      "spam.pow.numberOfTxPerBlock": "1000",
      "spam.protection.delegation.min.tokens": "100000000000000000",
      "spam.protection.max.batchSize": "30",
      "spam.protection.max.delegations": "360",
      "spam.protection.max.proposals": "300",
      "spam.protection.max.votes": "300",
      "spam.protection.maxUserTransfersPerEpoch": "10000",
      "spam.protection.minimumWithdrawalQuantumMultiple": "10",
      "spam.protection.minMultisigUpdates": "100000000000000000000",
      "spam.protection.proposal.min.tokens": "2000000000000000000000",
      "spam.protection.voting.min.tokens": "1000000000000000000",
      "spam.protection.applyReferral.min.funds": "0",
      "transfer.fee.factor": "0.001",
      "transfer.minTransferQuantumMultiple": "0",
      "transfer.fee.maxQuantumAmount": "100",
      "transfer.feeDiscountMinimumTrackedAmount": "0.001",
      "transfer.feeDiscountDecayFraction": "0.5",
      "validator.performance.scaling.factor": "0",
      "validators.delegation.minAmount": "1",
      "validators.epoch.length": "2m",
      "validators.vote.required": "0.67",
      "referralProgram.maxReferralTiers": "10",
      "referralProgram.maxReferralRewardFactor": "1",
      "referralProgram.maxReferralDiscountFactor": "1",
      "referralProgram.maxPartyNotionalVolumeByQuantumPerEpoch": "10000",
      "referralProgram.minStakedVegaTokens": "0",
      "referralProgram.maxReferralRewardProportion": "1",
      "market.aggressiveOrderBlockDelay": "1",
      "market.amm.minCommitmentQuantum": "1",
      "market.liquidity.maxAmmCalculationLevels": "100"
    },
    "network_limits": {
      "propose_market_enabled": true,
      "propose_asset_enabled": true,
      "bootstrap_block_count": 0
    },
    "checkpoint": {
      "load_hash": ""
    }
  }
}
```


### Relevant log output

```shell
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x10237d2b4]

goroutine 235134 [running]:
code.vegaprotocol.io/vega/datanode/api.(*TradingDataServiceV2).GetVolumeRebateStats(0x14002a2a000, {0x104200ac8, 0x14000c6b140}, 0x14003b5bd80)
	/Users/caed/Projects/vega-market-sim/extern/vega/datanode/api/trading_data_v2.go:5490 +0xd4
code.vegaprotocol.io/vega/protos/data-node/api/v2._TradingDataService_GetVolumeRebateStats_Handler.func1({0x104200ac8, 0x14000c6b140}, {0x103ed7980?, 0x14003b5bd80})
	/Users/caed/Projects/vega-market-sim/extern/vega/protos/data-node/api/v2/trading_data_grpc.pb.go:5320 +0x74
code.vegaprotocol.io/vega/datanode/ratelimit.(*RateLimit).GRPCInterceptor(0x14001ed8f20?, {0x104200ac8?, 0x14000c6b140?}, {0x103ed7980?, 0x14003b5bd80?}, 0x5000104?, 0x0?)
	/Users/caed/Projects/vega-market-sim/extern/vega/datanode/ratelimit/ratelimit.go:150 +0xcc
google.golang.org/grpc.getChainUnaryHandler.func1({0x104200ac8, 0x14000c6b140}, {0x103ed7980, 0x14003b5bd80})
	/Users/caed/go/pkg/mod/google.golang.org/grpc@v1.60.0/server.go:1192 +0xa0
code.vegaprotocol.io/vega/datanode/api.(*GRPCServer).Start.headersInterceptor.func4({0x104200ac8, 0x14000c6b140}, {0x103ed7980, 0x14003b5bd80}, 0x14000a0fae0?, 0x14003b5be00)
	/Users/caed/Projects/vega-market-sim/extern/vega/datanode/api/server.go:469 +0x328
google.golang.org/grpc.getChainUnaryHandler.func1({0x104200ac8, 0x14000c6b140}, {0x103ed7980, 0x14003b5bd80})
	/Users/caed/go/pkg/mod/google.golang.org/grpc@v1.60.0/server.go:1192 +0xa0
code.vegaprotocol.io/vega/datanode/api.(*GRPCServer).Start.(*GRPCServer).remoteAddrInterceptor.func3({0x104200ac8, 0x14000c6b0e0}, {0x103ed7980, 0x14003b5bd80}, 0x14000a0fae0, 0x14003b5bdc0)
	/Users/caed/Projects/vega-market-sim/extern/vega/datanode/api/server.go:425 +0x9c
google.golang.org/grpc.NewServer.chainUnaryServerInterceptors.chainUnaryInterceptors.func1({0x104200ac8, 0x14000c6b0e0}, {0x103ed7980, 0x14003b5bd80}, 0x140029499a8?, 0x100db42f8?)
	/Users/caed/go/pkg/mod/google.golang.org/grpc@v1.60.0/server.go:1183 +0x88
code.vegaprotocol.io/vega/protos/data-node/api/v2._TradingDataService_GetVolumeRebateStats_Handler({0x104187240?, 0x14002a2a000}, {0x104200ac8, 0x14000c6b0e0}, 0x14002ea6e80, 0x14002a12120)
	/Users/caed/Projects/vega-market-sim/extern/vega/protos/data-node/api/v2/trading_data_grpc.pb.go:5322 +0x12c
google.golang.org/grpc.(*Server).processUnaryRPC(0x14002a24000, {0x104200ac8, 0x14003dcba40}, {0x104216480, 0x14000bfb6c0}, 0x140025cd560, 0x14000ba2990, 0x1061a5638, 0x0)
	/Users/caed/go/pkg/mod/google.golang.org/grpc@v1.60.0/server.go:1372 +0xb8c
google.golang.org/grpc.(*Server).handleStream(0x14002a24000, {0x104216480, 0x14000bfb6c0}, 0x140025cd560)
	/Users/caed/go/pkg/mod/google.golang.org/grpc@v1.60.0/server.go:1783 +0xc4c
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	/Users/caed/go/pkg/mod/google.golang.org/grpc@v1.60.0/server.go:1016 +0x5c
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 442
	/Users/caed/go/pkg/mod/google.golang.org/grpc@v1.60.0/server.go:1027 +0x138
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -36,6 +36,7 @@
 - [11542](https://github.com/vegaprotocol/vega/issues/11542) - Fix non determinism in lottery ranking.
 - [11544](https://github.com/vegaprotocol/vega/issues/11544) - Fix empty candles stream.
 - [11579](https://github.com/vegaprotocol/vega/issues/11579) - Spot calculate fee on amend, use order price if no amended price is provided.
+- [11585](https://github.com/vegaprotocol/vega/issues/11585) - Initialise rebate stats service in API.
 
 
 ## 0.77.5

--- a/cmd/data-node/commands/start/node.go
+++ b/cmd/data-node/commands/start/node.go
@@ -242,6 +242,8 @@ func (l *NodeCommand) createGRPCServer(config api.Config) *api.GRPCServer {
 		l.timeWeightedNotionalPositionService,
 		l.gameScoreService,
 		l.ammPoolsService,
+		l.volumeRebateStatsService,
+		l.volumeRebateProgramService,
 	)
 	return grpcServer
 }

--- a/datanode/api/server.go
+++ b/datanode/api/server.go
@@ -194,6 +194,8 @@ type GRPCServer struct {
 	timeWeightedNotionalPositionService *service.TimeWeightedNotionalPosition
 	gameScoreService                    *service.GameScore
 	ammPoolService                      *service.AMMPools
+	volumeRebateStatsService            *service.VolumeRebateStats
+	volumeRebateProgramService          *service.VolumeRebatePrograms
 
 	eventObserver *eventObserver
 
@@ -266,6 +268,8 @@ func NewGRPCServer(
 	timeWeightedNotionalPositionService *service.TimeWeightedNotionalPosition,
 	gameScoreService *service.GameScore,
 	ammPoolService *service.AMMPools,
+	volumeRebateStatsService *service.VolumeRebateStats,
+	volumeRebateProgramsService *service.VolumeRebatePrograms,
 ) *GRPCServer {
 	// setup logger
 	log = log.Named(namedLogger)
@@ -337,7 +341,8 @@ func NewGRPCServer(
 		timeWeightedNotionalPositionService: timeWeightedNotionalPositionService,
 		gameScoreService:                    gameScoreService,
 		ammPoolService:                      ammPoolService,
-
+		volumeRebateStatsService:            volumeRebateStatsService,
+		volumeRebateProgramService:          volumeRebateProgramsService,
 		eventObserver: &eventObserver{
 			log:          log,
 			eventService: eventService,
@@ -578,6 +583,8 @@ func (g *GRPCServer) Start(ctx context.Context, lis net.Listener) error {
 		twNotionalPositionService:     g.timeWeightedNotionalPositionService,
 		gameScoreService:              g.gameScoreService,
 		AMMPoolService:                g.ammPoolService,
+		volumeRebateStatsService:      g.volumeRebateStatsService,
+		volumeRebateProgramService:    g.volumeRebateProgramService,
 	}
 
 	protoapi.RegisterTradingDataServiceServer(g.srv, tradingDataSvcV2)

--- a/datanode/api/trading_test.go
+++ b/datanode/api/trading_test.go
@@ -163,6 +163,8 @@ func getTestGRPCServer(t *testing.T, ctx context.Context) (tidy func(), conn *gr
 	gameScoreService := service.NewGameScore(sqlstore.NewGameScores(sqlConn), logger)
 	ammPoolsService := service.NewAMMPools(sqlstore.NewAMMPools(sqlConn))
 	sqlMarketDepthService := service.NewMarketDepth(service.NewDefaultConfig().MarketDepth, sqlOrderService, ammPoolsService, nil, nil, nil, nil, logger)
+	volumeRebateStatsService := service.NewVolumeRebateStats(sqlstore.NewVolumeRebateStats(sqlConn))
+	volumeRebateProgramssService := service.NewVolumeRebatePrograms(sqlstore.NewVolumeRebatePrograms(sqlConn))
 
 	g := api.NewGRPCServer(
 		logger,
@@ -225,6 +227,8 @@ func getTestGRPCServer(t *testing.T, ctx context.Context) (tidy func(), conn *gr
 		timeWeightedNotionPositionService,
 		gameScoreService,
 		ammPoolsService,
+		volumeRebateStatsService,
+		volumeRebateProgramssService,
 	)
 	if g == nil {
 		err = fmt.Errorf("failed to create gRPC server")

